packages/nx/src/tasks-runner/pseudo-terminal.ts

```diff
@@ -151,20 +156,37 @@ export class PseudoTtyProcess implements RunningTask {
 
   constructor(
     public rustPseudoTerminal: RustPseudoTerminal,
-    private childProcess: ChildProcess
+    private childProcess: ChildProcess,
+    private readyWhenStatus?: NormalizedRunCommandsOptions['readyWhenStatus']
   ) {
-    childProcess.onOutput((output) => {
-      this.terminalOutput += output;
-      this.outputCallbacks.forEach((cb) => cb(output));
-    });
+    this.initChildProcessesPromise = this.handleChildProcesses();
+  }
 
-    childProcess.onExit((message) => {
-      this.isAlive = false;
+  async handleChildProcesses() {
+    return new Promise((res) => {
+      this.childProcess.onOutput((output) => {
+        if (
+          this.readyWhenStatus?.length &&
+          isReady(this.readyWhenStatus, output)
+        ) {
+          this.isAlive = false;
+          this.childProcess.cleanup();
+          this.exitCallbacks.forEach((cb) => cb(0));
+          res({ success: true, code: 0, terminalOutput: this.terminalOutput });
+          return;
+        }
+        this.terminalOutput += output;
+        this.outputCallbacks.forEach((cb) => cb(output));
+      });
+
+      this.childProcess.onExit((message) => {
+        this.isAlive = false;
 
-      const code = messageToCode(message);
-      childProcess.cleanup();
+        const code = messageToCode(message);
+        this.childProcess.cleanup();
 
-      this.exitCallbacks.forEach((cb) => cb(code));
+        this.exitCallbacks.forEach((cb) => cb(code));
```

Comment:
We also need to resolve the promise here. This is needed in cases where the process exits before the readyWhen condition is met.